### DISCUSSION TITLE
Replace Papers Anti-xray with Orebfuscator or its Features

### DISCUSSION THREAD

**Discussion by Goinging** [PaperMC/Paper#11130] [Ideas]
### Is your feature request related to a problem?

Paper's anti-xray has problems with client side lag as well as other problems due to how it loads. Basically if a client is lagging loading chunks paper will think it is trying to xray and will display a mess of those blocks. Since the client has to load so many more block faces it lags even more causing a higher chance of this turning into a negative feedback loop. Throw in a laggy server as well and you run into even more problems. Paper is supposed to be the best server software out there when it comes to performance so this lack of care for client side performance is a problem.

### Describe the solution you'd like.

The [Orebfuscator Plugin](https://www.spigotmc.org/resources/orebfuscator-anti-x-ray.82710/) solves these problems and more. However, its features could also be implemented separately without adding in the whole plugin itself. Orebfuscator has dynamic block visibility and has a ton of [configuration options](https://github.com/Imprex-Development/Orebfuscator/wiki/Config) that allow for server owners to tailor the anti-xray best to their server. Key options include changing what blocks are displayed when the antixray is triggered and allowing the anti-xray to display in layers. With these changes you can remove a lot of the client side lag by removing air from displaying and when clients do lag having the blocks that display not being intrusive also helps. Ex. displaying netherack when the anti-ray is triggered in the nether when mining fast instead of fake ore. Orebfuscator does use Protocol Lib as a dependency so it may make more sense to only implement certain elements of Orebfuscator, however, the dynamic block visibility helps with client side lag a lot as well.

### Describe alternatives you've considered.

The best alternative is just to add many of these options to the paper anti-xray.

**Comment by kennytv** [comment]
There is no processing of what you can see, it simply mangles *all* hidden blocks, so this doesn't have anything to do with a laggy server or client. If you use the strongest mode, that means the client needs to load a lot of different types of blocks.

Most of what you described at the bottom we already have, see https://docs.papermc.io/paper/anti-xray for more info, otherwise you should bring up specific feature suggestions separately. Although you can of course simply use any other anti-xray plugin instead of the inbuilt options, being built into the server itself comes with advantages of performance and disadvantages of extendability, you need to weigh up whichever option is best for you
